apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/amqp_data_consumer.ex

```diff
@@ -164,57 +192,60 @@ defmodule Astarte.DataUpdaterPlant.AMQPDataConsumer do
   end
 
-  def handle_info({:async_init, queue_name}, _state) do
-    # This will block until Rabbit connection is up
-    rabbitmq_connection = ConnectionManager.get_connection()
+  defp schedule_connect() do
+    Process.send_after(self(), :connect, @reconnect_interval)
+  end
 
-    {:ok, chan} = initialize_chan(rabbitmq_connection, queue_name)
+  defp do_connect(state) do
+    conn = ExRabbitPool.get_connection_worker(:amqp_consumer_pool)
 
-    {:noreply, chan}
-  end
+    case ExRabbitPool.checkout_channel(conn) do
+      {:ok, channel} ->
+        _ =
+          Logger.debug(
+            "Successfully checked out channel for consumer on queue #{state.queue_name}"
+          )
 
-  def handle_info(
-        {:DOWN, _, :process, pid, :normal},
-        %Channel{pid: chan_pid, conn: %Connection{pid: conn_pid}} = chan
-      )
-      when pid != chan_pid and pid != conn_pid do
-    # This is a Message Tracker deactivating itself normally, do nothing
-    {:noreply, chan}
+        try_to_setup_consumer(channel, conn, state)
+
+      {:error, _reason} ->
+        _ = Logger.warn("Failed to check out channel for consumer on queue #{state.queue_name}")
+        schedule_connect()
+        {:noreply, state}
+    end
   end
 
-  # Make sure to handle monitored message trackers exit messages
-  # Under the hood DataUpdater calls Process.monitor so those monitor are leaked into this process.
-  def handle_info({:DOWN, _, :process, _pid, reason}, state) do
-    # Channel went down, stop the process
-    Logger.warn("AMQP channel crashed, reason: #{inspect(reason)}",
-      tag: "data_consumer_chan_crash"
-    )
+  defp try_to_setup_consumer(channel, conn, state) do
+    %Channel{pid: channel_pid} = channel
+    %State{queue_name: queue_name} = state
 
-    Channel.close(state)
-    {:stop, reason}
-  end
+    with :ok <- @adapter.qos(channel, prefetch_count: Config.consumer_prefetch_count!()),
+         {:ok, _queue} <- @adapter.declare_queue(channel, queue_name, durable: true),
+         {:ok, _consumer_tag} <- @adapter.consume(channel, queue_name, self()) do
+      ref = Process.monitor(channel_pid)
+
+      _ =
+        Logger.debug("AMQPDataConsumer for queue #{queue_name} initialized",
+          tag: "data_consumer_init_ok"
+        )
 
-  defp initialize_chan(rabbitmq_connection, queue_name) do
-    with {:ok, chan} <- Channel.open(rabbitmq_connection),
-         # Get a message if the channel goes down
-         Process.monitor(chan.pid),
-         :ok <- Basic.qos(chan, prefetch_count: Config.consumer_prefetch_count!()),
-         {:ok, _queue} <- Queue.declare(chan, queue_name, durable: true),
-         {:ok, _consumer_tag} <- Basic.consume(chan, queue_name) do
-      {:ok, chan}
+      {:noreply, %State{state | channel: channel, monitor: ref}}
     else
       {:error, reason} ->
         Logger.warn("Error initializing AMQPDataConsumer: #{inspect(reason)}",
           tag: "data_consumer_init_err"
         )
 
-        {:stop, reason}
```

Comment:
Probably the leak was happening here (if there was any kind of error the channel was opened and never closed)